source3/modules/vfs_glusterfs.c

```diff
@@ -152,6 +152,9 @@ static void glfs_clear_preopened(glfs_t *fs)
 			DLIST_REMOVE(glfs_preopened, entry);
 
 			glfs_fini(entry->fs);
+			
+			talloc_free(entry->connectpath);
+			talloc_free(entry->volume);
 			talloc_free(entry);
```

Comment:
The nice thing about talloc is the hierarchy:  since the connectpath and volume members are allocated on the parent context "entry", it is sufficient to call `talloc_free(entry);`